authprogs/authprogs.py

```diff
@@ -386,12 +386,93 @@ def find_match_scp(self, rule):  # pylint: disable-msg=R0911,R0912
             files = rule.get('files')
             if not isinstance(files, list):
                 files = [files]
-            if filepath not in files:
+            if rule.get('pcre_match'):
```

Comment:
Can we be more clear that this is a pcre against the *filename*.

Rather than overloading 'files' lets add a new one, path_pcre  (this will mirror the path_startswith that came in latest version)

    files:
      - foo.txt
      - bar.txt 
    paths_pcre: 
      - foo.*txt

Alternatively, if you don't actually need pcre and glob works then use `path_glob` - there is also now glob support in authprogs.py, which is used by rsync.